src/AddRecipe.js

```diff
@@ -0,0 +1,285 @@
+            title: this.state.title,
+            description: this.state.description,
+            calories: this.state.calories,
+            duration: this.state.duration,
+            imgURL: this.state.img,
+            ingredients: this.state.ingredients,
+            procedure: this.state.steps,
+            ratings: [0],
+            allergies: this.state.allergies,
+            genres: this.state.genres,
+            likes: 0,
+            timestamp: firebase.database.ServerValue.TIMESTAMP,
+        })
+
+        this.setState({
+            title: '',
+            description: '',
+            calories: '',
+            duration: 0,
+            imgURL: "",
+            ingredients: '',
+            instruction: '',
+            clock: -1,
+            steps: []
+        })
+
+        swal({
+            title: 'Recipe added!',
+            icon: 'success'
+        });
+    }
+
+    fileChange(event) {
+        // Get the uploaded file and its name
+        let name = event.target.files[0].name;
+        let file = event.target.files[0];
+
+        // Create a new child reference (on storage) for the image using its name
+        // Then, `put` the file contents. *then()*
+        // Using the `snapshot.downloadURL` value, push a new element into 
+        // The _database_ `imgs` reference
+        let imgRef = this.storageRef.child(name);
+        imgRef.put(file).then((snapshot) => {
+            this.dataRef.push({
+                url: snapshot.downloadURL
+            })
+
+            this.setState({
+                img: snapshot.downloadURL
+            })
+        });
+
+    }
+
+    addStep(event) {
+        let step = {
+            instruction: this.state.instruction,
+            clock: this.state.clock
+        }
+        this.state.steps.push(step);
+        this.clearStep();
+    }
+    
+    clearStep() {
+        this.setState({
+            instruction: '',
+            clock: -1
+        });
+        document.forms['add'].reset();
+    }
+
+    // renders input and button for a new post
+    render() {
+        return (
+            <div>
+                <h1>Add a Recipe</h1>
+                <form>
+                    <h2>Recipe Stats</h2>
+                    <input className="form-control mb-2" type="text" placeholder="Recipe Title" onChange={(event) => { this.setState({title: event.target.value}) }}/>
+                    <input className="form-control mb-2" type="text" placeholder="Short Description" onChange={(event) => { this.setState({description: event.target.value}) }}/>
+                    <input className="form-control mb-2" type="number" min="0" placeholder="Duration (minutes)" onChange={(event) => { this.setState({duration: event.target.value}) }}/>
+                    <input className="form-control mb-2" type="number" min="0" placeholder="Calorie Intake" onChange={(event) => { this.setState({calories: event.target.value}) }}/>
+                    
```

Comment:
Redundant - write a function to set the state based on the name of the input (we learned how to do this in class)